QuickBlox/q-municate-services-ios
# Add a `extendedRequest` parameter to `QMChatService`'s `messagesWithChatDialogID:completion:` and `earlierMessagesWithChatDialogID:completion:`

In our project we don't want QuickBlox to mark messages as read when we fetch them.

We use `QMChatService` and the`messagesWithChatDialogID:completion:` and `earlierMessagesWithChatDialogID:completion:` methods. But they don't allow us to provide our own extended request parameters (we need to fetch the messages with `mark_as_read=0`, see http://quickblox.com/developers/SimpleSample-chat_users-ios#List_chat_messages).

**Made/Proposed changes:**
- Added two more methods: `messagesWithChatDialogID:extendedRequest:completion:` and `earlierMessagesWithChatDialogID:extendedRequest:completion:`
- They set the defaults as in the previous implementation.
- Call the new methods from `messagesWithChatDialogID:completion:` and `earlierMessagesWithChatDialogID:completion:` respectively, with the `extendedRequest:nil`

**How should this be manually tested?**
- Check if `messagesWithChatDialogID:completion:` and `earlierMessagesWithChatDialogID:completion:` still work as expected.
- Check if the default request parameters are overridable when using `messagesWithChatDialogID:extendedRequest:completion:` and `earlierMessagesWithChatDialogID:extendedRequest:completion:`

**Does the documentation need an update?**
I've added the new methods to the header file and documented the parameters.
